### PR TITLE
Feature/Implement Router

### PR DESCRIPTION
**Pull request by bestwebua**
- [x] Implemented `Router` class
- [x] Updated `Application` class
- [x] Updated rspec config, added `RackHelpers`
- [x] Updated gemset

### CODE REVIEW

File: spec/support/helpers/rack_helpers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+RSpec.describe RackHelpers, type: :helper do
+  describe '#create_helper' do
+    subject(:helper) { create_rack_request(env) }
+
+    context 'without args' do
+      let(:env) { {} }
+
+      it 'returns Rack::Request instance double with default values' do
+        expect(helper).to be_an_instance_of(RSpec::Mocks::InstanceVerifyingDouble)
```
Comment (therealloker) [inline spec/support/helpers/rack_helpers_spec.rb]: Does this kind of expectations say something to us ?

File: spec/support/helpers/rack_helpers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        expect(helper.path).to eq('/')
+        expect(helper.params).to eq({})
+        expect(helper.env).to eq({})
+      end
+    end
+
+    context 'with args' do
+      let(:env) { { path: '/42', params: { a: 1 }, env: { b: 2 } } }
```
Comment (therealloker) [inline spec/support/helpers/rack_helpers_spec.rb]: may be this should have another name. env[:env] looks strange

File: spec/support/helpers/rack_helpers_spec.rb (hunk 1 above)
Comment (bestwebua) [reply spec/support/helpers/rack_helpers_spec.rb]: Yup. It's test for helper.

File: spec/support/helpers/rack_helpers_spec.rb (hunk 2 above)
Comment (bestwebua) [reply spec/support/helpers/rack_helpers_spec.rb]: Fixed.

**Pull request merged**